homeassistant/helpers/script.py

```diff
@@ -1218,6 +1229,17 @@ async def async_run(
         else:
             variables = cast(dict, run_variables)
 
+        # Prevent non-allowed recursive calls which will cause deadlocks when we try to
+        # stop (restart) or wait (queued) for our own script run.
+        script_stack = script_stack_cv.get()
+        if (
+            self.script_mode in (SCRIPT_MODE_RESTART, SCRIPT_MODE_QUEUED)
```

Comment:
Why is restart not allowed ?

Reply:
I guess if a script triggers itself with restart set, it will start itself, which then cancels current run, which then cancels the new started invocation of the script… 🤔 

yeah that's wrong.

Reply:
yeah, that's right 👍